tests/resources/generator/robertson_model_1966/model.dae.c

```diff
@@ -0,0 +1,103 @@
+/* The content of this file was generated using the C profile of libCellML 0.4.0. */
+
+#include "model.dae.h"
+
+#include <math.h>
+#include <stdlib.h>
+
+const char VERSION[] = "0.4.0";
+const char LIBCELLML_VERSION[] = "0.4.0";
+
+const size_t STATE_COUNT = 2;
+const size_t VARIABLE_COUNT = 5;
+
+const VariableInfo VOI_INFO = {"t", "dimensionless", "main", VARIABLE_OF_INTEGRATION};
+
+const VariableInfo STATE_INFO[] = {
+    {"y1", "dimensionless", "main", STATE},
+    {"y2", "dimensionless", "main", STATE}
+};
+
+const VariableInfo VARIABLE_INFO[] = {
+    {"k1", "dimensionless", "main", CONSTANT},
+    {"k3", "dimensionless", "main", CONSTANT},
+    {"y3", "dimensionless", "main", ALGEBRAIC},
+    {"k2", "dimensionless", "main", CONSTANT},
+    {"y2_scaled", "dimensionless", "main", ALGEBRAIC}
+};
+
+double * createStatesArray()
+{
+    return (double *) malloc(STATE_COUNT*sizeof(double));
+}
+
+double * createVariablesArray()
+{
+    return (double *) malloc(VARIABLE_COUNT*sizeof(double));
+}
+
+void deleteArray(double *array)
+{
+    free(array);
+}
+
+typedef struct {
+    double voi;
+    double *states;
+    double *rates;
+    double *variables;
+} RootFindingInfo;
+
+extern void nlaSolve(void (*objectiveFunction)(double *, double *, void *),
+                     double *u, int n, void *data);
+
+void objectiveFunction0(double *u, double *f, void *data)
+{
+    double voi = ((RootFindingInfo *) data)->voi;
+    double *states = ((RootFindingInfo *) data)->states;
+    double *rates = ((RootFindingInfo *) data)->rates;
+    double *variables = ((RootFindingInfo *) data)->variables;
```

Comment:
In the objective function the arrays that are not used are still declared, is this information that is not to hand at this point?

Reply:
Yeah, I must confess that I didn't bother generating code based on whether a given array is used or not. I guess we could do that later, if really deemed necessary, which I am not sure it is.